app/scripts/lib/ppom/ppom-middleware.ts

```diff
@@ -23,19 +27,33 @@ const ConfirmationMethods = Object.freeze([
  * the request will be forwarded to the next middleware, together with the PPOM response.
  *
  * @param ppomController - Instance of PPOMController.
+ * @param preferencesController - Instance of PreferenceController.
  * @returns PPOMMiddleware function.
  */
-export function createPPOMMiddleware(ppomController: PPOMController) {
+export function createPPOMMiddleware(
+  ppomController: PPOMController,
+  preferencesController: PreferencesController,
+) {
   return async (req: any, _res: any, next: () => void) => {
     try {
-      if (ConfirmationMethods.includes(req.method)) {
+      const securityAlertsEnabled =
+        preferencesController.store.getState()?.securityAlertsEnabled;
+      if (securityAlertsEnabled && ConfirmationMethods.includes(req.method)) {
         // eslint-disable-next-line require-atomic-updates
-        req.ppomResponse = await ppomController.usePPOM(async (ppom: PPOM) => {
-          return ppom.validateJsonRpc(req);
-        });
+        req.securityAlertResponse = await ppomController.usePPOM(
+          async (ppom: PPOM) => {
+            return ppom.validateJsonRpc(req);
+          },
+        );
       }
-    } catch (error: unknown) {
+    } catch (error: any) {
+      sentry?.captureException(error);
       console.error('Error validating JSON RPC using PPOM: ', error);
+      req.securityAlertResponse = {
+        result_type: BlockaidResultType.Failed,
+        reason: error.message,
+        description: 'Validating the confirmation failed by  throwing error.',
```

Comment:
just wondering, is this an arbitrary string we are providing? will it be used somewhere?

Reply:
maybe 
```suggestion
        description: 'JSON-RPC Validation failed. PPOM controller threw an error.',
```

Reply:
This is not used anywhere currently, just an additional piece of information.